hca_matrix_service/client.py

```diff
@@ -1,91 +1,83 @@
 import argparse
-import requests
 import zipfile
 import shutil
 import json
-import time
+import re
 import sys
 import os
+import urllib.request
+
 
 def parse_args():
-	parser = argparse.ArgumentParser("Download data via HCA's Matrix API V1. Requires either -p or -q input.")
-	parser.add_argument('-p', '--project', help="The project's Project Title, Project Label or link-derived ID, obtained from the HCA DCP, wrapped in quotes.")
-	parser.add_argument('-q', '--query', help="A complete /v1/matrix/ POST query in JSON format. Consult https://matrix.dev.data.humancellatlas.org/ for details.")
-	parser.add_argument('-f', '--format', default="loom", help="Format to download matrix in: loom, csv or mtx (Matrix Market). Defaults to loom.")
-	parser.add_argument('-o', '--outprefix', default=None, help="Output prefix for downloaded matrix. Leave default name (the Matrix API request ID) if not specified.")
-	args = parser.parse_args()
-	if not any(vars(args).values()):
-		parser.error('No arguments provided.')
-	return args
+    parser = argparse.ArgumentParser("Download data via HCA DCP FTP. Requires -p input. Files are downloaded into "
+                                     "current working directory.")
+    parser.add_argument('-p', '--project',
+                        help="The project's Project Title, Project short name or link-derived ID, obtained from the "
+                             "HCA DCP, wrapped in quotes.", required=True)
+    parser.add_argument('-f', '--format', default="loom", choices=["loom", "mtx"],
+                        help="Format to download matrix in: loom or mtx (Matrix Market). Defaults to loom.")
+    parser.add_argument('-o', '--outprefix', default=None,
+                        help="Output prefix to replace project uuid in filename of downloaded matrix. Leave as project "
+                             "uuid if not specified.")
+    args = parser.parse_args()
+    return args
+
+
+def load_project_index(path):
+    with urllib.request.urlopen(path) as response:
+        project_index = json.load(response)
+    return project_index
+
+def get_project_uuid(project_arg, project_index):
+    try:
+        # if uuid provided and it is in the index, return the uuid
+        if project_index[project_arg] and re.match('.{8}-.{4}-.{4}-.{4}-.{12}', project_arg):
+            print("Project uuid '{}' found with title '{}'.".format(project_arg, project_index[project_arg]
+                  ['project.project_core.project_title']))
+            return project_arg
+        # if not a uuid, look up the uuid and return it
+        print("Project '{}' found with uuid '{}'.".format(project_arg, project_index[project_arg]))
+        return project_index[project_arg]
+    except KeyError as e:
+        print(e)
+        print("The project identifier '{}' was not found in the database. Please check input and try again.".format(str(e)))
+        sys.exit()
```

Comment:
Please exit with an error code here, otherwise workflow environments/shell won't know that this failed.
```suggestion
        sys.exit(1)
```

Reply:
Added the suggestion!